src/bindings/python/flux/cli/base.py

```diff
@@ -1519,41 +1401,56 @@ def submit_cb(self, future, args, label=""):
             self.progress_update(submit_failed=True)
             return
 
-        if args.wait or args.watch:
-            #
-            #  If the user requested to wait for or watch all jobs
-            #   then start watching the main eventlog.
-            #
-            #  Carry along a bit of state for each job so that exceptions
-            #   before the job is running can be handled properly
-            #
-            jobinfo = {"id": jobid, "state": "submit"}
-            fut = job.event_watch_async(self.flux_handle, jobid)
-            fut.then(self.event_watch_cb, args, jobinfo, label)
-            self.progress_update(jobinfo, submit=True)
+        if self.watcher:
+            self.watcher.add_jobid(jobid, args.stdout, args.stderr, args.wait)
         elif self.progress:
             #  Update progress of submission only
             self.progress.update(jps=self.jobs_per_sec())
 
+    def _progress_check(self, args):
+        if args.progress and not self.progress and not sys.stdout.isatty():
+            LOGGER.warning("stdout is not a tty. Ignoring --progress option")
+            args.progress = None
+
+    def watcher_start(self, args):
+
+        if not self.watcher:
+            #  Need to open self.flux_handle if it isn't already in order
+            #  to start the watcher
+            if not self.flux_handle:
+                self.flux_handle = flux.Flux()
+
+            self._progress_check(args)
+
+            self.watcher = JobWatcher(
+                self.flux_handle,
+                progress=args.progress,
+                jps=args.jps,
+                log_events=(args.verbose > 1),
+                log_status=(args.verbose > 0),
+                labelio=args.label_io,
+                wait=args.wait,
+                watch=args.watch,
+            ).start()
+
     def jobs_per_sec(self):
         return (self.progress.count + 1) / self.progress.elapsed
 
     def progress_start(self, args, total):
         """
-        Initialize progress bar if one was requested
+        Initialize job submission progress bar user requested progress
+        without --wait or --watch
```

Comment:
need to nix second "progress" in this sentence?  Isn't reading quite right.

Reply:
Yeah, I think it is actually a missing "if", will fix.